stdlib/bool.mli

```diff
@@ -0,0 +1,65 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                         The OCaml programmers                          *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Boolean values.
+
+    @since 4.08 *)
+
+(** {1:bools Booleans} *)
+
+type t = bool
+(** The type for boolean values. *)
+
+val not : bool -> bool
+(** [not b] is the boolean negation of [b]. *)
+
+val negate : ('a -> bool) -> ('a -> bool)
+(** [negate sat v] is [not (sat v)]. *)
+
+external ( && ) : bool -> bool -> bool = "%sequand"
+(** [e0 && e1] is the lazy boolean conjunction of expressions [e0] and [e1].
+    If [e0] evaluates to [false], [e1] is not evaluated. Right-associative
+    operator at precedence level 3/11. *)
+
+external ( || ) : bool -> bool -> bool = "%sequor"
+(** [e0 || e1] is the lazy boolean disjunction of expressions [e0] and [e1].
+    If [e0] evaluates to [true], [e1] is not evaluated. Right-associative
+    operator at precedence level 2/11. *)
+
+(** {1:preds Predicates and comparisons} *)
+
+val equal : bool -> bool -> bool
+(** [equal b0 b1] is [true] iff [b0] and [b1] are both either [true]
+    or [false]. *)
+
+val compare : bool -> bool -> int
+(** [compare b0 b1] is a total order on boolean values. [false] is smaller
+    than [true]. *)
+
+(** {1:convert Converting} *)
+
+val to_int : bool -> int
+(** [to_int b] is [0] if [b] is [true] and [1] if [b] is [false]. *)
+
+val to_float : bool -> float
+(** [to_float b] is [0.] if [b] is [true] and [1.] if [b] is [false]. *)
+
+val of_string : string -> bool option
+(** [of_string s] is [Some true] if [s] is ["true"], [Some false] if [s]
+    is ["false"] and [None] otherwise. *)
```

Comment:
It might be useful to also accommodate different capitalizations of "true" and "false", perhaps by doing `lowercase_ascii` over the value before the comparison.

Reply:
That would break backward compatibility w.r.t. deprecation path.

Reply:
Then it shouldn't be done. It would be nice if there was a standard function that did this, though. Perhaps also add a function called `of_string_case_fold` following the precedent in `Str`?

Reply:
These kind of things are always a bit endless why not also map "0" to "false" etc. It's preferable to leave that to proper parsing libraries.

Reply:
It's probably best to choose a good (or observe the existing) textual representation, use it for `to_string`, and define `of_string` minimally so that `X.equal x (of_string (to_string x))`, irrespective of what `X` is. IIUC, that is what this currently does.

Reply:
We are indeed presuming `of_string` will remain minimal. The question is whether additional functions might be of use. But we can always add them later.

I tend to favor adding such functions because otherwise people just re-implement them over and over again. If you look at OCaml programs, many of them come along with the creator's favorite kit of things that should be in the standard library. In the long run, it would be nice if the standard library was rich enough with common small functions that this rarely happened.